js/util/util.js

```diff
@@ -41,6 +41,14 @@ exports.wrap = function (n, min, max) {
     return n === max ? n : ((n - min) % d + d) % d + min;
 };
 
+exports.coalesce = function() {
```

Comment:
can we pick a less intimidating name for this? Maybe `util.pick` or `util.oneOf`...